[MSVC][permissive-] Ppsspp failed to build with msvc due to error C2039, error C2065...... on Windows

### Game or games this happens in

None

### What area of the game / PPSSPP

Hi all,
PPsspp fails to build on MSVC due to error C2039, error C2065, error C2923 error C2976, error C2641, error C2371, error C2955, error C2780. Could you help look?

we use commit is 91886be.

**Repro steps**
1. git clone https://github.com/hrydgard/ppsspp F:\gitP\hrydgard\ppsspp
2. set _CL_= /permissive-
3. mkdir build_amd64 & cd F:\gitP\hrydgard\ppsspp\build_amd64
4. msbuild /m /p:Platform=x64 /p:Configuration=Release ../Windows/PPSSPP.sln /t:Rebuild /p:OutDir=F:\gitP\hrydgard\ppsspp\build_amd64 
**error info**
F:\gitP\hrydgard\ppsspp\ext\armips\Parser\DirectivesParser.h(16,51): error C2039: 'string': is not a member of 'std' (compiling source file armips\Parser\DirectivesParser.cpp) [F:\gitP\hrydgard\ppsspp\ext\libarmips.vcxproj]
F:\gitP\hrydgard\ppsspp\ext\armips\Parser\DirectivesParser.h(16,57): error C2065: 'string': undeclared identifier (compiling source file armips\Parser\DirectivesParser.cpp) [F:\gitP\hrydgard\ppsspp\ext\libarmips.vcxproj]
F:\gitP\hrydgard\ppsspp\ext\armips\Parser\DirectivesParser.h(16,27): error C2923: 'std::unordered_multimap': 'string' is not a valid template type argument for parameter '_Kty' (compiling source file armips\Parser\DirectivesParser.cpp) [F:\gitP\hrydgard\ppsspp\ext\libarmips.vcxproj]
F:\gitP\hrydgard\ppsspp\ext\armips\Parser\DirectivesParser.h(16,27): error C2976: 'std::unordered_multimap': too few template arguments (compiling source file armips\Parser\DirectivesParser.cpp) [F:\gitP\hrydgard\ppsspp\ext\libarmips.vcxproj]


### What should happen

Build successfully

### Logs

[build (13).log](https://github.com/hrydgard/ppsspp/files/9868325/build.13.log)


### Platform

Windows

### Mobile phone model or graphics card

None

### PPSSPP version affected

None

### Last working version

None

### Graphics backend (3D API)

Other

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [ ] Try resetting settings or older versions and include if the issue is related.
- [ ] Try without any cheats and without loading any save states.
- [ ] Include logs or screenshots of issue.

--- a/Common/GPU/thin3d.h
+++ b/Common/GPU/thin3d.h
@@ -393,6 +393,9 @@ struct AutoRef {
 		*this = p.ptr;
 		return *this;
 	}
+	bool operator !=(const AutoRef<T> &p) const {
+		return ptr != p.ptr;
+	}
 
 	T *operator->() const {
 		return ptr;

--- a/UWP/Common/DeviceResources.cpp
+++ b/UWP/Common/DeviceResources.cpp
@@ -470,13 +470,13 @@ void DX::DeviceResources::SetWindow(CoreWindow^ window)
 			try
 			{
 				const auto dm = hdi->GetCurrentDisplayMode();
-				const unsigned int hdmi_width = dm->ResolutionWidthInRawPixels;
-				const unsigned int hdmi_height = dm->ResolutionHeightInRawPixels;
+				const float hdmi_width = (float)dm->ResolutionWidthInRawPixels;
+				const float hdmi_height = (float)dm->ResolutionHeightInRawPixels;
 				// If we're running on Xbox, use the HDMI mode instead of the CoreWindow size.
 				// In UWP, the CoreWindow is always 1920x1080, even when running at 4K.
 
 				m_logicalSize = Windows::Foundation::Size(hdmi_width, hdmi_height);
-				m_dpi = currentDisplayInformation->LogicalDpi * 1.5;
+				m_dpi = currentDisplayInformation->LogicalDpi * 1.5f;
 			}
 			catch (const Platform::Exception^)
 			{

--- a/ext/native/tools/atlastool/atlastool.vcxproj
+++ b/ext/native/tools/atlastool/atlastool.vcxproj
@@ -97,7 +97,6 @@
       <ConformanceMode>true</ConformanceMode>
       <AdditionalIncludeDirectories>../../../../;../../ext;../../;../../../../ext;../prebuilt;../../../../ext/zstd/lib</AdditionalIncludeDirectories>
       <RuntimeLibrary>MultiThreaded</RuntimeLibrary>
-      <BufferSecurityCheck>false</BufferSecurityCheck>
     </ClCompile>
     <Link>
       <SubSystem>Console</SubSystem>

